src/index.ts

```diff
@@ -50,10 +50,34 @@ export const temporal = (<TState>(
         temporalStateCreator(set, get, options),
     );
 
-    const curriedHandleSet =
-      options?.handleSet?.(
-        (store.temporal.getState() as _TemporalState<TState>)._handleSet,
-      ) || (store.temporal.getState() as _TemporalState<TState>)._handleSet;
+    const userHandleSet = options?.handleSet?.(
+      (store.temporal.getState() as _TemporalState<TState>)
+        ._handleSet as StoreApi<TState>['setState'],
+    );
+
+    const internalHandleSet = (
+      store.temporal.getState() as _TemporalState<TState>
+    )._handleSet;
+
+    const curriedHandleSet = userHandleSet || internalHandleSet;
```

Comment:
For my understanding, these lines here are a more verbose way to write these four lines, correct?

```ts
   const curriedHandleSet =
      options?.handleSet?.(
        (store.temporal.getState() as _TemporalState<TState>)._handleSet,
      ) || (store.temporal.getState() as _TemporalState<TState>)._handleSet;
```

Reply:
If so, could we revert these changes to cut down on the number of lines changed? Thanks!